src/traces/image/plot.js

```diff
@@ -82,48 +95,132 @@ module.exports = function plot(gd, plotinfo, cdimage, imageLayer) {
+                    } else {
+                        // Return a transparent pixel
+                        context.fillStyle = 'rgba(0,0,0,0)';
+                    }
+                    context.fillRect(ipx0, jpx0, ipx1 - ipx0, jpx1 - jpx0);
                 }
-                context.fillRect(ipx0, jpx0, ipx1 - ipx0, jpx1 - jpx0);
             }
+
+            return canvas;
+        }
+
+        function sizeImage(image) {
+            image.attr({
+                height: imageHeight,
+                width: imageWidth,
+                x: left,
+                y: top
+            });
         }
 
+        var data = (trace._isFromSource && !fastImage) ? [cd, {hidden: true}] : [cd];
         var image3 = plotGroup.selectAll('image')
-            .data(cd);
+            .data(data);
 
         image3.enter().append('svg:image').attr({
             xmlns: xmlnsNamespaces.svg,
             preserveAspectRatio: 'none'
         });
 
-        image3.attr({
-            height: imageHeight,
-            width: imageWidth,
-            x: left,
-            y: top,
-            'xlink:href': canvas.toDataURL('image/png')
+        if(fastImage) sizeImage(image3);
+        image3.exit().remove();
+
+        // Pixelated image rendering
+        // http://phrogz.net/tmp/canvas_image_zoom.html
+        // https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering
+        image3
+            .attr('style', 'image-rendering: optimizeSpeed; image-rendering: -moz-crisp-edges; image-rendering: -o-crisp-edges; image-rendering: -webkit-optimize-contrast; image-rendering: optimize-contrast; image-rendering: crisp-edges; image-rendering: pixelated;');
+
+        var p = new Promise(function(resolve) {
+            if(trace._isFromZ) {
+                resolve();
+            } else if(trace._isFromSource) {
+                // Check if canvas already exists
+                if(
+                    trace._canvas &&
+                    trace._canvas.el.width === w &&
+                    trace._canvas.el.height === h &&
+                    trace._canvas.source === trace.source
+                ) {
+                    resolve();
+                } else {
+                    // Create a canvas and transfer image onto it to access pixel information
+                    var canvas = document.createElement('canvas');
+                    canvas.width = w;
+                    canvas.height = h;
+                    var context = canvas.getContext('2d');
+
+                    var sel;
+                    if(fastImage) {
+                        // Use the displayed image
+                        sel = image3;
+                    } else {
+                        // Use the hidden image
+                        sel = d3.select(image3[0][1]);
+                    }
+
+                    var image = sel.node();
+                    image.onload = function() {
+                        context.drawImage(image, 0, 0);
```

Comment:
This lines throws an error in the console in Chrome `v <= 58`.

Reply:
That raises an interesting question: how far back do we support Chrome?

Reply:
Our statement has generally been that aside from IE we support current versions of browsers. Chrome 58 is over 3 years old, so I'd say no, we don't need to support this.